strawberry/annotation.py

```diff
@@ -182,11 +187,61 @@ def create_union(self, evaled_type: Type) -> StrawberryUnion:
             return evaled_type
 
         types = evaled_type.__args__
+
         union = StrawberryUnion(
             type_annotations=tuple(StrawberryAnnotation(type_) for type_ in types),
         )
+
+        union_args = [arg for arg in args if isinstance(arg, StrawberryUnion)]
+        if len(union_args) > 1:
+            logging.warning(
+                "Duplicate union definition detected. "
+                "Only the first definition will be considered"
+            )
+
+        if union_args:
+            arg = union_args[0]
+            union.graphql_name = arg.graphql_name
+            union.description = arg.description
+            union.directives = arg.directives
+
+            union._source_file = arg._source_file
+            union._source_line = arg._source_line
+
+        self.validate_union_members(types, union)
+
         return union
 
+    def validate_union_members(
+        self, types: Tuple[type], union: StrawberryUnion
+    ) -> None:
+        from strawberry.union import StrawberryUnion
+
+        scalars = (int, str, float)
+
+        for type_ in types:
+            # Handle case: x = Annotated[Union[X, Y], strawberry.union("X")]
+            if get_origin(type_) is Annotated:
+                # Unwrap annotated type into the proper type hints
+                # and our strawberry type metadata
+                inner_type, *metadata = get_args(type_)
+                union_members = get_args(inner_type)
+
+                union_definition = (
+                    metadata[0]
+                    if metadata and isinstance(metadata[0], StrawberryUnion)
+                    else None
+                )
+
+                for member in union_members:
+                    if isinstance(member, scalars):
```

Comment:
Also, it is not just those scalars (int, str and float) that cannot be used. We also have custom scalars which would need to be checked.

Maybe the best way in here is to check that the members are strawberry types (i.e. probably by using `has_object_definition`?) and not inputs because inputs cannot be used in unions as well.

Reply:
I removed this code, we already had another place that was doing a similar check 😊